src/strategyHelpers/deploymentHelper.ts

```diff
@@ -59,17 +61,14 @@ export async function deployManifests(
          const routeStrategy = parseRouteStrategy(
             core.getInput('route-method', {required: true})
          )
-
-         const {workloadDeployment, newObjectsList} = await Promise.resolve(
-            (routeStrategy == RouteStrategy.INGRESS &&
-               deployBlueGreenIngress(kubectl, files)) ||
-               (routeStrategy == RouteStrategy.SMI &&
-                  deployBlueGreenSMI(kubectl, files, annotations)) ||
-               deployBlueGreenService(kubectl, files)
+         const {deployResult} = await deployBlueGreen(
+            kubectl,
+            files,
+            routeStrategy
          )
 
-         checkForErrors([workloadDeployment.result])
-         return workloadDeployment.newFilePaths
+         checkForErrors([deployResult.result])
```

Comment:
I feel like "deployment" is misleading because the resource wrapping the "deployResult" field is called a BlueGreenDeployment but I'll rename the field to be "execResult" since it's more specific/makes clear the fact that the thing being extracted is just a specific result within this larger "result"